packages/itwin/property-grid/src/components/PropertyGridContent.tsx

```diff
@@ -93,13 +94,36 @@ export function PropertyGridContent({
 
   const [filterText, setFilterText] = useState<string>("");
   const { showNullValues } = useNullValueSettingContext();
+  const { onFeatureUsed } = useTelemetryContext();
   const filterer = useFilterer({ showNullValues, filterText });
 
+  const filterTextRef = useLatest(filterText);
+  const showNullValuesRef = useLatest(showNullValues);
+
   const [{ width, height }, setSize] = useState({ width: 0, height: 0 });
   const handleResize = useCallback((w: number, h: number) => {
     setSize({ width: w, height: h });
   }, []);
 
+  const reportFilterFeature = useCallback(() => {
+    if (filterTextRef.current.length > 0) {
+      onFeatureUsed?.("filter-properties");
+    }
+  }, [onFeatureUsed, filterTextRef]);
+
+  useEffect(() => {
+    reportFilterFeature();
+  }, [filterText, reportFilterFeature]);
+
+  useEffect(() => {
+    return dataProvider.onDataChanged.addListener(() => {
+      if (!dataProvider.keys.isEmpty) {
+        onFeatureUsed?.(showNullValuesRef.current ? "hide-empty-values-disabled" : "hide-empty-values-enabled");
```

Comment:
I think this should be done in `useNullValueSettingContext`

Reply:
As discussed, added a `onDataLoaded` callback to `AutoExpandingPropertyFilterDataProvider` and used it to report this setting.